src/Mod/Sketcher/App/Sketch.cpp

```diff
@@ -1425,6 +1425,22 @@ std::vector<Part::Geometry *> Sketch::extractGeometry(bool withConstructionEleme
     return temp;
 }
 
+GeoListFacade Sketch::extractGeoListFacade() const
+{
+    std::vector<GeometryFacadeUniquePtr> temp;
+    temp.reserve(Geoms.size());
+    int internalGeometryCount = 0;
+    for (std::vector<GeoDef>::const_iterator it=Geoms.begin(); it != Geoms.end(); ++it) {

```

Comment:
Prefer a ranged for loop if possible

Reply:
I must admit I am still conflicted by the "prefer range loop". 

Until now I have adopted the view that it is a matter of style and preference.

We don't have containers with copy-on-write semantics (to my knowledge), so I guess we could opt for range loops. Even with them, it is still possible to use std::as_const (c++17) to get away with it. However, the strong argument for readability no longer holds when one needs to write ```for(const auto & v : std::as_const(container))```.

Let me think about it.